dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -977,11 +984,12 @@ void Sedp::rekey_volatile(const Security::SPDPdiscoveredParticipantData& pdata)
 
 #endif // OPENDDS_SECURITY
 
-void Sedp::disassociate_helper(const BuiltinEndpointSet_t& avail, const CORBA::ULong flags,
+void Sedp::disassociate_helper(BuiltinEndpointSet_t& avail, const CORBA::ULong flags,
                                const RepoId& id, const EntityId_t& ent, DCPS::TransportClient& client)
 {
   if (avail & flags) {
     client.disassociate(make_id(id, ent));
+    avail ^= flags;
```

Comment:
This is incorrect.

Reply:
I was wondering about that... is there a need to change `avail` at all here?